copy/copy.go

```diff
@@ -615,13 +613,16 @@ func (c *copier) copyConfig(ctx context.Context, src types.Image) error {
 		if err != nil {
 			return errors.Wrapf(err, "Error reading config blob %s", srcInfo.Digest)
 		}
-		bar := c.createProgressBar(srcInfo, "config")
-		defer bar.Finish()
-		bar.Start()
+
+		// make the short digest only 10 characters long to make it align with the blob output
+		bar := c.createProgressBar(srcInfo, shortDigest(srcInfo.Digest)[0:10], "config")
 		destInfo, err := c.copyBlobFromStream(ctx, bytes.NewReader(configBlob), srcInfo, nil, false, true, bar)
 		if err != nil {
 			return err
 		}
+		if bar != nil {
```

Comment:
`bar` can never be `nil` AFAICS, so drop this check. (The code would crash in `bar.ProxyReader` anyway.)

Reply:
There was one test case where this happened. I could not figure out why and also had a look at the mpb code but figured to just add the check to defend from the corner case. I think it's related to non-TTY as the test case (i.e., `TestCopyOCIRoundTrip`) worked when doing it manually.

Reply:
Second, working around crashes without understanding how they happen is not great for reliability. AFAICS the only way `bar` can be `nil` is if `AddBar` returns `nil`, which in turn can happen only if `Progress.Wait` has been called.

… which probably points to 
https://github.com/containers/image/pull/581#discussion_r257475486 , calling `Wait` and looping to `copyUpdatedConfigAndManifest` as the underlying cause: the code tried schema2, failed, tried schema1, and `copyConfig` did not use the `bar` (and did not trigger the `bar.ProgressReader` crash) because schema1 does not have a separate config object.

I’ll take a stab at fixing this; if you can remember what the failing situation was (or is it just straightforward `make test-skopeo` failing?), that would be useful.

Reply:
You're absolutely right, and I was/am not happy with it either. The failure is straightforward to reproduce with integration tests (i.e., `TestCopyOCIRoundTrip`).

I think a more or less easy solution is to break out the parts printing (including related code) and move the Wait() call to the end of copyOneImage.

Reply:
Thanks, reproduced. I was not quite right about this happening with schema1 fallback:it is copying an OCI image and falling back to schema2.  For the record, the reason it does not crash in `bar.ProgressReader` is that creating the `ProgressReader` does not access `bar`, only non-empty reads do, and `PutBlob` detects that the config is already present at the destination, so no reads happen.

I think I know how to deal with this…

Reply:
> I think I know how to deal with this…

Thank you so much for looking into this :pray: